### ISSUE TITLE
Creating onhour for private instance

### ISSUE THREAD

**Issue by srinivas-anant** [capitalone/cloud-custodian#1614]
I have created a onhour policy to start the instance at a particular time but the instance is private instance it doesn't seem to be turning on, but when I apply the same policy to a public instance it works proper. 

Is there a option to do it for private instances or the feature is not available yet?

**Comment by srinivas-anant** [comment]
@kapilt

**Comment by kapilt** [comment]
there really isn't any distinguishing between private vs public for the onhour policy, it just cares there's an instance with the given tag.. i'm confused as to the behavior your seeing.

**Comment by srinivas-anant** [comment]
@kapilt - I found the issue, when I checked the resource.json file in the output folder 
```
"StateReason": {
      "Message": "Client.InternalError: Client error on launch", 
      "Code": "Client.InternalError"
    }, 
```
This is the error message it kept showing, it was not able to launch the instance because the IAM credentials that cloud custodian is using to start the EC2 instance may not have permissions to access the KMS key used to decrypt the root EBS volume.
I updated the IAM policy for my AWS credentials to add the "kms:CreateGrant" command.

And it worked. 

This is resource I refereed to --> https://skeddly.desk.com/customer/en/portal/articles/2757270-ec2-instance-starts-but-immediately-stops
